### ISSUE TITLE
Fixing consistency of "Left" and "Right" oligos on Variation and PCR Oligo pages

### ISSUE THREAD

**Issue by wormbasehelp** [WormBase/website#6437]
*Submitted from the feedback form on the WormBase website.*

<p>&nbsp;&nbsp;The left and right Oligo of gk448 in Variation is different from those in PCR Product/Oligo, which one is correct?</p>

**Reported by:** MM (6465************)
**Submitted from:** <a target="_blank" href="http://www.wormbase.org//tools/support">/tools/support</a>
**Browser:** Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36

**Comment by chris-grove** [comment]
Response to user:
"""
Thanks for reporting this issue. It appears that what is considered "Left Oligo" and "Right Oligo" has been flipped/reversed on the variation page:


https://www.wormbase.org/species/c_elegans/variation/WBVar00145853#05678--10


compared to the respective PCR Oligo pages:


https://www.wormbase.org/species/all/pcr_oligo/gk448_external#0--10


https://www.wormbase.org/species/all/pcr_oligo/gk448_internal#0--10


So, technically, both primer sets are correct, it's just that the "Left"/"Right" labels have been swapped between the two different pages. In this case, the PCR Oligo pages show the correct "Left"/"Right" labels when considering the default genome browser view and the orientation of the affected gene, pdr-1. The "Left"/"Right" labels on the Variation page could be considered correct if viewing from the reverse orientation, but I agree this inconsistency is confusing and should be corrected. I will notify the appropriate people to get this fixed.


If you ever want to test primers electronically, you can use the ePCR tool to predict where primers would amplify in the C. elegans genome:


https://wormbase.org/tools/epcr


You can, of course, flip the "Left" and "Right" oligos and still get the same PCR product, as left and right is relative.


You can follow progress on the issue on GitHub here:


https://github.com/WormBase/website/issues/6437
"""

**Comment by a8wright** [comment]
@chris-grove I am assigning this one to myself. I was planning on converting this one over for the next release. Although, this widget requires a lot of code. I will make sure this fix is implemented with the conversion of the widget!

**Comment by chris-grove** [comment]
@a8wright 
OK thanks!!

**Comment by a8wright** [comment]
@chris-grove I came across this ticket yesterday. Unfortunately, the fix did not go out when I created the widget. I have now implemented a fix. The code now determines the left and right oligo sequence for the molecular details widget the same way as the pcr_oligo overview widget. They should now always match. The change is currently on staging. If you like the change I will push it out to production.
